src/system.rs

```diff
@@ -128,6 +130,384 @@ pub struct LoadAverage {
     pub last_pid: PID,
 }
 
+#[derive(Debug, Clone)]
+pub struct CpuTimes {
+    /// Time spent by normal processes executing in user mode;
+    /// on Linux this also includes guest time
+    pub user: u64,
+
+    /// Time spent by niced (prioritized) processes executing in user mode;
+    /// on Linux this also includes guest_nice time
+    pub nice: u64,
+
+    /// Time spent by processes executing in kernel mode
+    pub system: u64,
+
+    /// Time spent doing nothing
+    pub idle: u64,
+
+    /// Time spent waiting for I/O to complete
+    pub iowait: u64,
+
+    /// Time spent for servicing hardware interrupts
+    pub irq: u64,
+
+    /// Time spent for servicing software interrupts
+    pub softirq: u64,
+
+    /// Time spent by other operating systems running in a virtualized environment
+    pub steal: u64,
+
+    /// Time spent running a virtual CPU for guest operating systems
+    /// under the control of the Linux kernel
+    pub guest: u64,
+
+    /// Time spent running a niced guest
+    /// (virtual CPU for guest operating systems
+    /// under the control of the Linux kernel)
+    pub guest_nice: u64,
+}
+
+impl CpuTimes {
+    /// Initialize CpuTimes struct
+    pub fn new(
```

Comment:
Is there any need for a `new()` method on the structs if the method does no processing?

Reply:
I added it by coherence with the existing code but I didn't check that the existing new() methods had additional processing.